### ISSUE TITLE
The JSXExpression union incorrectly includes JSXEmptyExpression

### ISSUE THREAD

**Issue by MageJohn** [typescript-eslint/typescript-eslint#6315]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

ast-spec

### Playground Link

https://typescript-eslint.io/play/#ts=4.3.5&jsx=true&sourceType=module&showAST=es&code=DwQwfA3gvsD04G4g&eslintrc=N4KABGBEBOCuA2BTAzpAXGYBfEWg&tsconfig=N4KABGBEDGD2C2AHAlgGwKYCcDyiAuysAdgM6QBcYoEEkJemy0eAcgK6qoDCAFutAGsylBm3TgwAXxCSgA

### Repro Code

```typescript
// This parses as a JSXExpressionContainer node whose expression is a JSXEmptyExpression node.
<a>{}</a>;
```

### Expected Result

I expect that when using the `AST_NODE_TYPES.JSXExpression` type the `AST_NODE_TYPES.JSXEmptyExpression` won't be a member of the union.

### Actual Result

What I see is that the definition of the union is:

```tsx
export type JSXExpression =
  | JSXEmptyExpression
  | JSXExpressionContainer
  | JSXSpreadChild;
```

### Additional Info

Code like `<a>{}</a>` is parsed by typescript as a `JsxExpression` node who's `expression` value is `undefined`. As part of converting the JsxExpression node to an ESLint compatible AST, the `typescript-estree` package replaces the `undefined` with a `JSXEmptyExpression` node ([relevant code](https://github.com/typescript-eslint/typescript-eslint/blob/863e43ea161d5c0bf72d9de2f869bf61e6383eec/packages/typescript-estree/src/convert.ts#L2329-L2348)). 

However, in the `ast-spec` package the `JSXEmptyExpression` type is a part of the `JSXExpression` union ([defined here](https://github.com/typescript-eslint/typescript-eslint/blob/863e43ea161d5c0bf72d9de2f869bf61e6383eec/packages/ast-spec/src/unions/JSXExpression.ts)). As explained above, this is incorrect as the `JSXEmptyExpression` node is only created as the child of a JSXExpression. 

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/ast-spec` | `5.48.0` |
| `@typescript-eslint/typescript-estree` | `5.48.0` |

**Comment by bradzacher** [comment]
From the looks of our snapshots, yeah I think that `JSXEmptyExpression` should only be valid as a child. Of `JSXExpressionContainer`.

I thought there was a case that you could have an empty expression by itself, but I can't seem to find it.
